### PR TITLE
Dockerize my app

### LINKED ISSUE #64: Create Your Own Toolkit for Goose! 🛠️
# Create Your Own Toolkit for Goose! 🛠️

[Goose](https://github.com/block/goose) extends its capabilities through toolkits. Share your custom toolkit to help other developers automate their workflows!

## 🎯 Goal
Create a custom toolkit for Goose and contribute it to our [community plugins](https://github.com/block-open-source/goose-plugins/tree/main/src/goose_plugins/toolkits).

## 🔑 Tasks and Acceptance Criteria
- [ ] Design and implement your toolkit
- [ ] Test your toolkit locally
- [ ] Write tests for your toolkit -- see here for an [example](https://github.com/block-open-source/goose-plugins/blob/main/tests/toolkits/test_todo.py)
- [ ] Add your toolkit to the goose-plugins repository
- [ ] Create a pull request with your changes

Your toolkit should:
- Solve a specific development workflow challenge
- Include clear documentation
- Follow Goose's toolkit structure
- Include tests where appropriate

## 🎓 Resources
- [Goose Documentation](https://block.github.io/goose/)
- [Goose repo](https://github.com/block/goose)
- [How to use toolkits](https://dev.to/blackgirlbytes/how-my-ai-dev-agent-turns-screenshots-into-ui-updates-189p)
- [How to create toolkits](https://dev.to/blackgirlbytes/how-to-extend-your-ai-agent-for-custom-developer-workflows-4fld)

## Getting Started
1. Comment ".take" on this issue to get assigned
2. Fork the repository and create a new branch
3. Develop your toolkit
4. Submit a PR adding your toolkit to the `/src/goose_plugins/toolkits` directory
5. Wait for review and address any feedback

## Questions?
Need help or clarification? Comment on this issue!

Happy coding! 🦢

### PR DESCRIPTION
**Pull request by Johnnyevans32**
# Dockerization Toolkit (Dockerize-My-App) for Goose

# Resolves #64 

## Overview
The **Dockerization Toolkit** automates the creation of Docker-related configuration files for software projects. It detects the project type and generates essential Docker configuration files such as `Dockerfile`, `.dockerignore`, and optionally `docker-compose.yml` for multi-container setups.

---

## Features
- **Automatic Project Detection**:
  - **Python**: Detects `requirements.txt` or `pyproject.toml`.
  - **Node.js**: Detects `package.json`.
  - **Java**: Detects `pom.xml` or `build.gradle`.
- **Generated Files**:
  - **Dockerfile**: Optimized configuration for each project type.
  - **.dockerignore**: Excludes unnecessary files from the Docker build context.
  - **docker-compose.yml** (optional): Configures multi-container orchestration.
---

## Usage
1. Add the toolkit to your `profiles.yaml`:

```yaml
my_profile:
  provider: openai
  processor: gpt-4o
  accelerator: gpt-4o-mini
  moderator: passive
  toolkits:
    - dockerize_my_app
```

2. Run Goose with the profile:

```bash
goose session start --profile my-profile
```

3. Use the toolkit:

```bash
dockerize my app  <path_to_project> into <output_directory>
```

---

## Output
The toolkit generates the following files in the specified output directory:

#### For Python Projects:
```dockerfile
FROM python:3.11-slim
WORKDIR /app
COPY requirements.txt .
RUN pip install --no-cache-dir -r requirements.txt
COPY . .
CMD ["python", "app.py"]
```

#### For Node.js Projects:
```dockerfile
FROM node:18-alpine
WORKDIR /app
COPY package*.json ./
RUN npm install
COPY . .
CMD ["npm", "start"]
```

#### For Java Projects:
```dockerfile
FROM openjdk:17-jdk-slim
WORKDIR /app
COPY . .
RUN ./gradlew build
CMD ["java", "-jar", "build/libs/app.jar"]
```

### **.dockerignore**
```dockerignore
.git/
.env
node_modules/
__pycache__/
*.pyc
target/
```

### **docker-compose.yml** (if applicable):
```yaml
version: "3.8"
services:
  app:
    build:
      context: .
    ports:
      - "8080:8080"
```

---

## Testing
Run tests using `pytest`:

```bash
uv run pytest tests -m "not integration"
```

---

### **`pyproject.toml`**
```toml
[project.entry-points."goose.toolkit"]
dockerize_my_app = "goose.toolkit.dockerize_my_app:DockerizationToolkit"
```

### CODE REVIEW

**Comment by Johnnyevans32** [comment]
fixed now

File: .gitignore [hunk 1]
```diff
@@ -27,6 +27,9 @@ share/python-wheels/
 *.egg
 MANIFEST
 
+
+.DS_Store
```
Comment (blackgirlbytes) [inline .gitignore:31]: hmm should we remove this since it wasnt in the original gitignore?

File: pyproject.toml [hunk 2]
```diff
@@ -6,7 +6,7 @@ readme = "README.md"
 requires-python = ">=3.10"
 dependencies = [
     "ai-exchange>=0.8.4",
-    "goose-ai>=0.9.8",
+    "goose-ai>=0.9.8"
```
Comment (blackgirlbytes) [inline pyproject.toml:9]:
```suggestion
    "goose-ai>=0.9.8",
```

File: .gitignore (hunk 1 above)
Comment (Johnnyevans32) [reply .gitignore]: fixed now @blackgirlbytes

**Review by blackgirlbytes — APPROVED** [review]
looking good! Thanks for the contribution!

**Pull request merged**